### ISSUE TITLE
[Bug]: Update nb_inventory to work with NetBox 3.5

### ISSUE THREAD

**Issue by rodvand** [netbox-community/ansible_modules#987]
### Ansible NetBox Collection version

v3.12.0

### Ansible version

```
ansible [core 2.13.9]
```

### NetBox version

v3.5-beta2

### Python version

3.8

### Steps to Reproduce

Use nb_inventory plugin with NetBox 3.5.

### Expected Behavior

It to present a working inventory.

### Observed Behavior

Crashes on fetching the openapi schema,

**Comment by nzurku** [comment]
Didn't realize this broke until we started trying netbox 3.5 in a new staging area. The inventory files were still referencing an earlier netbox instance until I fixed it and this broke.

**Comment by MrXermon** [comment]
Seems like the OpenAPI path moved from "/api/docs/?format=openapi" to "/api/schema/".

**Comment by salman79uk** [comment]
Is there a fix planned for this?

**Comment by eric-eisenhart** [comment]
I did a bit of digging into this.

TL;DR: It will take more than just changing what URL nb_inventory pulls.

/api/docs/?format=openapi changes to /api/schema/?format=json but also changes from "swagger 2.0" to "openapi 3.0.3". This adds "/api/" to the start of paths and might have other changes.

Most of /api/docs/ moved to /api/schema/swagger-ui/ but that appears to be (basically) a javascript app to explore the API.

demo.netbox.dev is currently netbox 3.5 and can be used to check what's returned: https://demo.netbox.dev/api/schema/?format=json

**netbox 3.4 /api/docs/?format=openapi:**
```json
{
  "swagger": "2.0",
  "info": {
    "title": "NetBox API",
    "description": "API to access NetBox",
    "termsOfService": "https://github.com/netbox-community/netbox",
    "license": {
      "name": "Apache v2 License"
    },
    "version": "3.4"
  },
#...
  "paths": {
#...
    "/dcim/devices/": {
      "get": {
        "operationId": "dcim_devices_list",
        "description": "",
        "parameters": [
#...
```
  
**netbox 3.5 /api/schema/?format=json:**
```json
{
  "openapi": "3.0.3",
  "info": {
      "title": "NetBox API",
      "version": "3.5.0 (3.5)",
      "description": "API to access NetBox",
      "license": {
          "name": "Apache v2 License"
      }
  },
  "paths": {
 #...
       "/api/dcim/devices/": {
          "get": {
            "operationId": "dcim_devices_list",
            "description": "Get a list of device objects.",
            "parameters": [
```

**Comment by jiuka** [comment]
For us it works with the following changes to the inventory plugin. https://github.com/scsitteam/netbox_ansible_modules/commit/9d0e992a7d3ffbe7a9bd1b2fe4f3b0f37efd4722

**Comment by salfers** [comment]
As a server-side workaround you can take the API schema from a Netbox 3.4 instance (like [this](https://paste.debian.net/plainh/51ea947b)) and configure your web server to serve it for the request:
```
    rewrite ^/api/docs/ /static/dummy-openapi.json break;
```
Then put the file into e.g. `/opt/netbox/netbox/static/dummy-openapi.json`.

I haven't found any downsides to this and it avoids the inconvenience of patching multiple Ansible installations.

**Comment by eric-eisenhart** [comment]
FYI: Netbox v3.5.1 released a few hours ago and changes the returned paths to not have the `/api/` prefix anymore, so a rewrite rule might now be enough.

**Comment by rodvand** [comment]
Included in #999

**Comment by adparis99** [comment]
Hi there! I upgraded to netbox.netbox 3.13 using the `ansible-galaxy collection install netbox.netbox` command, but I'm still getting an error when trying to render the inventory using `ansible-inventory -i --list`. This is a different, shorter error output than what I was getting before. See below for the current (post 3.13 update) vs prior (pre 3.13 update) message:
Error Output 1:
`
[WARNING]:  * Failed to parse netbox_inventory_config.yml with auto plugin:  
<!DOCTYPE html>
<html lang="en">  
<head>     
<title>Server Error</title>
     <link rel="stylesheet" href="/static/netbox-light.css" />
<meta charset="UTF-8"> 
</head>  
<body>     
	<div class="container-fluid">
		<div class="row">             
			<div class="col col-md-6 offset-md-3">
				<div class="card border-danger mt-5">
					<h5 class="card-header"><i class="mdi mdi-alert"></i>Server Error</h5>
					<div class="card-body">
						<p>
							There was a problem with your request. Please contact an administrator.                             
						</p>
						<hr />
						<p>The complete exception is provided below:
						</p> <pre class="block"><strong>&lt;class &#x27;AttributeError&#x27;&gt;</strong><br /> &#x27;cached_property&#x27; object has no attribute &#x27;items&#x27;  
						Python version: 3.10.6 NetBox version: 3.5.1</pre>
						<p>If further assistance is required, please post to the <a href="https://github.com/netbox-community/netbox/discussions">NetBox discussion forum </a> on GitHub.
						</p>
						<div class="text-end">
							<a href="/" class="btn btn-primary">Home Page</a>
						</div>
					</div>
				</div>
			</div>
		</div>
	</div>
</body>
</html>
`
Error Output 2:
[WARNING]:  * Failed to parse /opt/ansible/netbox_inventory_config.yml with auto plugin:     <!DOCTYPE html> <html   lang="en"   data-netbox-url-name=""   data-netbox-base-path=""               data-netbox-color-mode="unset"
>   <head>     <meta charset="UTF-8" />     <meta       name="viewport"       content="initial-scale=1, maximum-scale=1, user-scalable=no, width=device-width, viewport-fit=cover"     />           <title>Page Not Found |
NetBox</title>      <script       type="text/javascript"       src="/static/setmode.js"       onerror="window.location='/media-failure/?filename=setmode.js'">     </script>      <script type="text/javascript">       (function () {
initMode()       })();       window.CSRF_TOKEN = "bSrnfoIOfZ8lpTUo31Oxf9bX6vHgKLjkL4OuYtc2NtP4W9f5bllv4BEu9f7TQot1";     </script>           <link       rel="stylesheet"       href="/static/netbox-external.css?v=3.5.1"
onerror="window.location='/media-failure/?filename=netbox-external.css'"     />     <link       rel="stylesheet"       href="/static/netbox-light.css?v=3.5.1"       onerror="window.location='/media-failure/?filename=netbox-
light.css'"     />     <link       rel="stylesheet"       href="/static/netbox-dark.css?v=3.5.1"       onerror="window.location='/media-failure/?filename=netbox-dark.css'"     />     <link       rel="stylesheet"       media="print"
href="/static/netbox-print.css?v=3.5.1"       onerror="window.location='/media-failure/?filename=netbox-print.css'"     />     <link rel="icon" type="image/png" href="/static/netbox.ico" />     <link rel="apple-touch-icon"
type="image/png" href="/static/netbox_touch-icon-180.png" />           <script       type="text/javascript"       src="/static/netbox.js?v=3.5.1"       onerror="window.location='/media-failure/?filename=netbox.js'">     </script>
</head>    <body>     <script type="text/javascript">       function checkSideNav() {         // Check localStorage to see if the sidebar should be pinned.         var sideNavRaw = localStorage.getItem('netbox-sidenav');         //
Determine if the device has a small screeen. This media query is equivalent to         // bootstrap's media-breakpoint-down(lg) breakpoint mixin, which is what the sidenav's         // CSS uses.         var isSmallScreen =
window.matchMedia('(max-width: 991.98px)').matches;         if (typeof sideNavRaw === 'string') {           var sideNavState = JSON.parse(sideNavRaw);           if (sideNavState.pinned === true && !isSmallScreen) {             //
If the sidebar should be pinned and this is not a small screen, set the appropriate             // body attributes prior to the rest of the content rendering. This prevents             // jumpy/glitchy behavior on page reloads.
document.body.setAttribute('data-sidenav-pinned', '');             document.body.setAttribute('data-sidenav-show', '');             document.body.removeAttribute('data-sidenav-hidden');           } else {
document.body.removeAttribute('data-sidenav-pinned');             document.body.setAttribute('data-sidenav-hidden', '');           }         }       }       window.addEventListener('resize', function(){ checkSideNav() });
checkSideNav();     </script>               <div class="container-fluid px-0">     <main class="layout">             <nav class="sidenav noprint" id="sidenav" data-simplebar>   <div class="sidenav-header">               <a
class="sidenav-brand" href="/">       <img src="/static/netbox_logo.svg" height="48" class="sidenav-brand-img" alt="NetBox Logo">     </a>           <a class="sidenav-brand-icon" href="/">       <img src="/static/netbox_icon.svg"
height="32" class="sidenav-brand-img" alt="NetBox Logo">     </a>           <button class="btn btn-sm btn-ghost-primary sidenav-toggle">       <div class="sidenav-toggle-icon">         <i class="mdi mdi-pin"></i>       </div>
</button>    </div>    <div class="sidenav-inner h-100 mb-auto">           <div class="collapse sidenav-collapse pb-4">                 <ul class="navbar-nav">        <li class="nav-item">              <a class="nav-link"
href="#menutopology-views" data-bs-toggle="collapse" role="button" aria-expanded="false" aria-controls="menutopology-views">        <i class="mdi mdi-sitemap"></i>        <span class="nav-link-text">Topology Views</span>       </a>
<div class="collapse" id="menutopology-views">         <ul class="nav nav-sm flex-column">                                     <li class="nav-item">               <div class="nav-group-header">                 <span class="nav-
group-label">TOPOLOGY</span>               </div>             </li>                                         <li class="nav-item no-buttons">                                  <a href="/plugins/netbox_topology_views/topology/"
class="nav-link">Topology</a>                                                 </li>                                                  <li class="nav-item">               <div class="nav-group-header">                 <span
class="nav-group-label">PREFERENCES</span>               </div>             </li>                                         <li class="nav-item no-buttons">                                  <a
href="/plugins/netbox_topology_views/images/" class="nav-link">Images</a>                                                 </li>                            <li class="nav-item no-buttons">                                  <a
href="/plugins/netbox_topology_views/individualoptions/" class="nav-link">Individual Options</a>                                                 </li>                                 </ul>       </div>     </li>        <li
class="nav-item">              <a class="nav-link" href="#menuPlugins" data-bs-toggle="collapse" role="button" aria-expanded="false" aria-controls="menuPlugins">        <i class="mdi mdi-puzzle"></i>        <span class="nav-link-
text">Plugins</span>       </a>              <div class="collapse" id="menuPlugins">         <ul class="nav nav-sm flex-column">                                     <li class="nav-item">               <div class="nav-group-header">
<span class="nav-group-label">NextBox UI</span>               </div>             </li>                                         <li class="nav-item no-buttons">                                  <a href="/plugins/nextbox-
ui/topology/" class="nav-link">Topology Viewer</a>                                                 </li>                                 </ul>       </div>     </li>    </ul>       </div>           <div class="profile-button-
container">          <div class="btn-group">     <a class="btn btn-primary ws-nowrap" type="button" href="/login/?next=/api/docs/">       <i class="mdi mdi-login-variant"></i> Log In     </a>     <button type="button" class="btn
btn-primary dropdown-toggle dropdown-toggle-split" data-bs-toggle="dropdown">       <span class="visually-hidden">Toggle Dropdown</span>     </button>     <ul class="dropdown-menu dropdown-menu-end">       <li>         <button
class="dropdown-item color-mode-toggle">           <i class="color-mode-icon mdi mdi-lightbulb"></i>           <span class="color-mode-text">Dark Mode</span>         </button>       </li>     </ul>   </div>       </div>   </div>
</nav>                <div class="content-container" tabindex="-2">                   <div class="p-2 printonly">           <img src="/static/netbox_logo.svg" alt="NetBox logo" width="200px" />         </div>                   <nav
class="navbar navbar-light sticky-top flex-md-nowrap p-1 mb-3 search container-fluid noprint">                           <div class="nav-mobile">               <div class="nav-mobile-top">                 <a class="sidebar-logo p-2
d-block" href="/">                   <img src="/static/netbox_logo.svg" alt="NetBox logo" height="40" />                 </a>                 <button type="button" aria-label="Toggle Navigation" class="navbar-toggler sidenav-
toggle-mobile">                   <span class="navbar-toggler-icon"></span>                 </button>               </div>               <div class="d-flex my-1 flex-grow-1 justify-content-center w-100">                 <form
class="input-group" action="/search/" method="get">   <input name="q" type="text" aria-label="Search" placeholder="Search" class="form-control" />   <button class="btn btn-primary" type="submit">     <i class="mdi mdi-magnify"></i>
</button> </form>                </div>             </div>                           <div class="row search-container">                               <div class="col-3 d-flex flex-grow-1 ps-0"></div>
<div class="col-6 d-flex flex-grow-1 justify-content-center">                 <form class="input-group" action="/search/" method="get">   <input name="q" type="text" aria-label="Search" placeholder="Search" class="form-control" />
<button class="btn btn-primary" type="submit">     <i class="mdi mdi-magnify"></i>   </button> </form>                </div>                               <div class="col-3 d-flex flex-grow-1 pe-0 justify-content-end">
<div class="btn-group">     <a class="btn btn-primary ws-nowrap" type="button" href="/login/?next=/api/docs/">       <i class="mdi mdi-login-variant"></i> Log In     </a>     <button type="button" class="btn btn-primary dropdown-
toggle dropdown-toggle-split" data-bs-toggle="dropdown">       <span class="visually-hidden">Toggle Dropdown</span>     </button>     <ul class="dropdown-menu dropdown-menu-end">       <li>         <button class="dropdown-item
color-mode-toggle">           <i class="color-mode-icon mdi mdi-lightbulb"></i>           <span class="color-mode-text">Dark Mode</span>         </button>       </li>     </ul>   </div>                 </div>              </div>
</nav>                                                            <div class="title-container px-3 pb-3">                           <div id="content-title">                              <h1 class="h2 w-100">Page Not Found</h1>
</div>                                       </div>                            <div id="content" class="container-fluid content px-0 m-0">                                   <div class="px-3">                <div class="row my-5">
<div class="col-6 offset-3">         <div class="card">             <h5 class="card-header text-danger">                 /api/docs/             </h5>             <div class="card-body">                      The requested page does
not exist.              </div>             <div class="card-footer text-end">                 <a href="/" class="btn btn-sm btn-primary">Home Page</a>             </div>         </div>     </div> </div>              </div>
</div>                                      <footer class="footer container-fluid">                        <div class="row align-items-center justify-content-between mx-0">                <div class="col-sm-12 col-md-auto fs-4
noprint">                 <nav class="nav justify-content-center justify-content-lg-start">                                                             <a type="button" class="nav-link" href="/static/docs/" target="_blank">
<i title="Docs" class="mdi mdi-book-open-variant text-primary" data-bs-placement="top" data-bs-toggle="tooltip"></i>                     </a>                                           <a type="button" class="nav-link" href="/api/"
target="_blank">                       <i title="REST API" class="mdi mdi-cloud-braces text-primary" data-bs-placement="top" data-bs-toggle="tooltip"></i>                     </a>                                           <a
type="button" class="nav-link" href="/api/schema/swagger-ui/" target="_blank">                       <i title="REST API documentation" class="mdi mdi-book text-primary" data-bs-placement="top" data-bs-toggle="tooltip"></i>
</a>                                                                <a type="button" class="nav-link" href="/graphql/" target="_blank">                       <i title="GraphQL API" class="mdi mdi-graphql text-primary" data-bs-
placement="top" data-bs-toggle="tooltip"></i>                     </a>                                                                <a type="button" class="nav-link" href="https://github.com/netbox-community/netbox"
target="_blank">                       <i title="Source Code" class="mdi mdi-github text-primary" data-bs-placement="top" data-bs-toggle="tooltip"></i>                     </a>                                           <a
type="button" class="nav-link" href="https://netdev.chat/" target="_blank">                       <i title="Community" class="mdi mdi-slack text-primary" data-bs-placement="top" data-bs-toggle="tooltip"></i>
</a>                                    </nav>               </div>                <div class="col-sm-12 col-md-auto text-center text-lg-end text-muted">                 <span class="d-block d-md-inline"><span title="May 17, 2023
9:16 p.m.">2023-05-17 21:16</span> UTC</span>                 <span class="ms-md-3 d-block d-md-inline">netbox (v3.5.1)</span>               </div>              </div>                    </footer>        </div>      </main>
</div>                         <div id="django-messages" class="toast-container">                </div>            <div id="netbox-data" style="display: none!important; visibility: hidden!important">            </div>    </body>
</html>

**Comment by rnwolfe** [comment]
@adparis99 you might try `ansible-galaxy collection install netbox.netbox --force` to ensure the new version is installed. It didn't fix for me without the `--force`.
